crates/fluvio-protocol-derive/src/util.rs

```diff
@@ -1,75 +1,256 @@
                 }
+
+                Ok(PropAttrsType::Lit(syn::Ident::new(value, lit.span())))
+            } else {
+                Err(syn::Error::new(
+                    span,
+                    format!("Expected {attr_name} attribute to be a LitStr or LitInt: `{attr_name} = \"...\"`"),
+                ))
             }
         }
+        Some(Expr::Unary(ExprUnary { expr, .. })) => {
+            // When passing -1 as a value it is returned as type Unary
+            // So to handle that we are checking if it's Unary Lit and continue as usual
+            // If needed this can be extended to handle the Unary operators
+            // But it doesn't seem that is necessary currently
+            Ok(PropAttrsType::Int(parse_int_expr(attr_name, expr, span)?))
+        }
+        Some(Expr::Path(ExprPath { path, .. })) => {
+            // For now we only need Path just to handle cases where CONSTANTS are being passed without "" quotes
+            if let Some(path_ident) = path.get_ident() {
+                return Ok(PropAttrsType::Lit(syn::Ident::new(
+                    &path_ident.to_string(),
+                    path_ident.span(),
+                )));
+            }
+
+            Err(syn::Error::new(
+                span,
+                format!("Attribute {attr_name} needs to be a valid Path: `{attr_name} = \"...\"`"),
+            ))
+        }
+        _ => Err(syn::Error::new(
+            span,
+            format!(
+                "Expected {attr_name} attribute to be a Lit/Path/Unary: `{attr_name} = \"...\"`"
+            ),
+        )),
     }
+}
 
-    None
+fn parse_int_expr<'a>(attr_name: &'a str, expr: &'a Expr, span: Span) -> syn::Result<i16> {
+    match expr {
+        Expr::Unary(ExprUnary {
+            op: UnOp::Neg(_),
+            expr,
+            ..
+        }) => parse_int_expr(attr_name, expr, span).map(|int| -int),
+        Expr::Lit(ExprLit {
+            lit: Lit::Int(int), ..
+        }) => int.base10_parse(),
+        _ => Err(syn::Error::new(
+            span,
+            format!("Expected {attr_name} to be Int: `{attr_name} = \"...\"`"),
+        )),
+    }
 }
 
-/// find name value with integer value
-pub(crate) fn find_int_name_value(version_meta: &Meta, attr_name: &str) -> Option<u64> {
-    if let Some(attr) = find_name_attribute(version_meta, attr_name) {
-        match &attr.lit {
-            Lit::Int(version_val) => {
-                //  println!("version value: {}",version_val.value());
-                version_val.base10_parse::<u64>().ok()
+pub fn get_lit_int_value<'a>(
+    attr_name: &'a str,
+    value: &'a Option<Expr>,
+    span: Span,
+) -> syn::Result<i16> {
+    if let Some(value_expr) = value {
+        parse_int_expr(attr_name, value_expr, span)
+    } else {
+        Err(syn::Error::new(
+            span,
+            format!("Expected {attr_name} to be valid Expr: `{attr_name} = \"...\"`"),
+        ))
+    }
+}
+pub fn get_lit_int<'a>(
```

Comment:
We have two functions here `get_lit_int` and `get_lit_int_value` that do the same thing. Can we make one from them that can cover both cases?

Reply:
I kept them both but improved a bit on them. Also it was incorrect to call it `get_lit_int_value`. I renamed it to `get_expr_int_value` because even though we are returning Int value we need to check for the Unary operator as well and returns just the int value, that's also why I kept them both because the get_lit_int only does checks for int without the Unary and returns Lit.